docs/reference/snapshot-restore/apis/get-snapshot-status-api.asciidoc

```diff
@@ -0,0 +1,266 @@
+Number of shards that failed to be included in the snapshot.
+
+`total`::
+(integer)
+Total number of shards included in the snapshot.
+====
+
+[[get-snapshot-status-stats]]
+`stats`::
+(object)
+Provides details on the number (`file_count`) and size (`size_in_bytes`) of files included in the snapshot.
++
+.Properties of `stats`
+[%collapsible%open]
+====
+`incremental`::
+(object)
+Number and size of files that are not already in the repository and need to be copied as part of the incremental snapshot.
+
+`processed`::
+(object)
+Number and size of files that are still being copied.
+
+`total`::
+(object)
+Total number and size of files that are referenced by the snapshot.
+
+`start_time_in_millis`::
+(long)
+The time, in milliseconds, when the snapshot creation process started.
+
+`time_in_millis`::
+(long)
+The total time, in milliseconds, that
+====
+
+`indices`::
+(list of objects)
+List of objects containing information about the
+indices included in the snapshot.
++
+.Properties of `indices`
+[%collapsible%open]
+====
+`shards_stats`::
+See <<get-snapshot-status-shards-stats,`shards_stats`>>.
+
+`stats`::
+See <<get-snapshot-status-stats,`stats`>>.
+
+`shards`::
+(list of objects)
+List of objects containing information about the
+shards that include the snapshot.
++
+.Properties of `shards`
+[%collapsible%open]
+=====
+`stage`::
+(string)
+Indicates the current state of the shards that include the snapshot.
++
+.Properties of `stage`
+[%collapsible%open]
+======
+`DONE`::
+Number of shards containing a completed copy of the snapshot.
+
+`FAILURE`::
+Number of shards containing a failed copy of the snapshot.
+
+`FINALIZE`::
+Number of shards containing a copy of the snapshot in the finalizing stage.
+
+`INIT`::
+Number of shards containing a copy of the snapshot in the initializing stage.
+
+`STARTED`::
+Number of shards containing a copy of the snapshot in the started stage.
+======
+=====
+====
+
+[[get-snapshot-status-api-example]]
+==== {api-example-title}
+
+The following request returns information for `snapshot_2` in the `my_repository` repository.
+
+[source,console]
+----
+GET /_snapshot/my_repository/snapshot_2/_status
+----
+
+The API returns the following response:
+
+[source,console-result]
+----
+{
+  "snapshots" : [
+    {
+      "snapshot" : "snapshot_2",
+      "repository" : "my_repository",
+      "uuid" : "ITWHQY2fR2OqX-TYt6mUNA",
+      "state" : "SUCCESS",
+      "include_global_state" : true,
+      "shards_stats" : {
+        "initializing" : 0,
+        "started" : 0,
+        "finalizing" : 0,
+        "done" : 0,
+        "failed" : 0,
+        "total" : 0
+      },
+      "stats" : {
+        "incremental" : {
+          "file_count" : 0,
+          "size_in_bytes" : 0
+        },
+        "total" : {
+          "file_count" : 0,
+          "size_in_bytes" : 0
+        },
+        "start_time_in_millis" : 1594158151205,
+        "time_in_millis" : 0
+      },
+      "indices" : { }
+    }
+  ]
+}
+----
```

Comment:
It'd be good to create a couple of indices and add some documents to them so this is a little more realistic.

Currently, all the stats are zero and there are no indices, which is where a good bit of the stats are located.

Reply:
Good call 👍 I'm adding a few indices locally to generate tangible figures for those properties. I'll update the examples with that data.